### DISCUSSION TITLE
Create cable connections in bulk

### DISCUSSION THREAD

**Discussion by shatt79** [netbox-community/netbox#9286] [Q&A]
Is there any way to make cable connections in bulk?  I have 288ct (and larger) fiber panels that connect to multiple smaller panels.  I know I could create multi-position rear ports and just connect rear ports together, but I have hundreds of panels and almost as many configurations.  It'd be faster to be able to select 48 rear ports and connect those to 48 rear ports on panel X.  Then take 144 rear ports and connect those to panel Y.  Then 2x24 counts may go to two other panels, etc, etc.

**Comment by andy-lawson** [comment]
I've used cable imports for 24 port patch panels - would that scale up?
I create the CSVs in Excel and populate as required, then import into Netbox.
Much easier than linking PP rear ports together individually.

**Reply by jeremystretch** [reply]
This is the recommended approach. You could also create a simple custom script to populate cables between sets of ports.

**Reply by candlerb** [reply]
Like this one:
https://github.com/netbox-community/reports/blob/master/scripts/multi_connect.py

**Reply by shatt79** [reply]
I'll take a look at both options.  Thanks!

**Reply by shatt79** [reply]
> Like this one: https://github.com/netbox-community/reports/blob/master/scripts/multi_connect.py

This script doesn't seem to work in 3.2.   I get this error:

<class 'ValueError'>

too many values to unpack (expected 2)

Python version: 3.8.10
NetBox version: 3.2.0

I'm not a coder, so I don't know how to fix.

**Reply by candlerb** [reply]
Needs a backtrace, but I'm not sure how you can get a backtrace from a custom script. Setting `DEBUG = True` in configuration.py might help, but I haven't tested this.

**Reply by trs91km** [reply]
Here also if I gave two interfaces, its failing
xe-1-1,xe-1-0 are the two interfaces of my devices


failure message: Unable to find "xe-1-1,xe-1-0" in interfaces on device B (test.com)

**Reply by candlerb** [reply]
I don't think a comma-separated list of interface names is allowed here. But a pattern like `xe-1-[0-1]` should be.

It's whatever Netbox's [expand_alphanumeric_pattern](https://github.com/netbox-community/netbox/blob/v3.5.3/netbox/utilities/forms/utils.py#L79-L90) allows.
